### PR TITLE
[Enchance] Support sync random seed for distributed sampler

### PR DESCRIPTION
**Pull request by linyq17**
Modify distributed sampler and keep the manual_seed behavior same with pytorch original version

Refers to https://github.com/open-mmlab/mmdetection/pull/7440/files, https://github.com/open-mmlab/mmdetection/pull/7432/files, and https://github.com/open-mmlab/mmediting/pull/815.

### CODE REVIEW

File: mmfewshot/utils/infinite_sampler.py [hunk 1]
```diff
@@ -28,7 +30,13 @@ def __init__(self,
                  seed: int = 0,
                  shuffle: bool = True) -> None:
         self.dataset = dataset
-        self.seed = seed if seed is not None else 0
+        # In distributed sampling, different ranks should sample
```
Comment (jshilong) [inline mmfewshot/utils/infinite_sampler.py:33]: where do we set this seed outside

File: mmfewshot/utils/dist_utils.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import torch.distributed as dist
+from mmcv.runner import get_dist_info
+
+
+def check_dist_init():
+    return dist.is_available() and dist.is_initialized()
+
+
+def sync_random_seed(seed=None, device='cuda'):
+    """Make sure different ranks share the same seed.
```
Comment (jshilong) [inline mmfewshot/utils/dist_utils.py:13]: add an extra comment  
Propagating the seed of rank 0 to all other ranks

**Comment by jshilong** [comment]
LGTM
